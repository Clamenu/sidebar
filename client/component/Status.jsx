import React from 'react';
import StatusHoursContainer from '../container/StatusHoursContainer.jsx';
import StatusPriceRangeContainer from '../container/StatusPriceRangeContainer.jsx';
import StatusHealthContainer from '../container/StatusHealthContainer.jsx';
import styles from '../sidebar.css';

const Status = () => (
  <div className={styles.statusContainer}>
    <table>
      <tbody>
        <StatusHoursContainer />
        <StatusPriceRangeContainer />
        <StatusHealthContainer />
      </tbody>
    </table>
  </div>
);

export default Status;
