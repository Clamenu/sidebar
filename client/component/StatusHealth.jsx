import React from 'react';
import styles from '../sidebar.css';

const StatusHealth = ({ healthScore }) => (
  <tr className={styles.statusIconRow}>
    <td><img src='images/health.png' className={styles.statusHealth}/></td>
    <td className={styles.statusScore}>
      <b className={styles.healthScoreLink}>Health Score</b>
      <b id='health-score'>&nbsp;&nbsp;{healthScore} out of 100</b>
    </td>
  </tr>
);

export default StatusHealth;