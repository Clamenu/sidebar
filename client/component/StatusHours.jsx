import React from 'react';
import PropTypes from 'prop-types';
import styles from '../sidebar.css';

const StatusHours = ({ hours, isOpen }) => (
  <tr>
    <td className={styles.statusIconRow}>
      {isOpen
        ? <img src="images/openClock.png" className={styles.statusClock} alt="#" />
        : <img src="images/closeClock.png" className={styles.statusClock} alt="#" />}
    </td>
    <td className={styles.statusHours} id="status-hours">
      Today&nbsp;&nbsp;
      {isOpen
        ? (
          <b>
            {hours.split(', ')[0]}
            &nbsp;&nbsp;
          </b>
        )
        : (
          <b>
            {hours.split(', ')[0]}
            <br />
          </b>
        )}
      {isOpen && hours.split(', ')[1] && (
        <b>
          <br />
          {hours.split(', ')[1]}
          &nbsp;&nbsp;
        </b>
      )}
      {!isOpen && hours.split(', ')[1] && (
        <b>
          {hours.split(', ')[1]}
          &nbsp;&nbsp;
        </b>
      )}
      {isOpen
        ? <span className={styles.open}>Open now</span>
        : <span className={styles.close}>Closed now</span>}
    </td>
    <td className={styles.emptyRow} />
  </tr>
);

export default StatusHours;

StatusHours.propTypes = {
  hours: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
};
