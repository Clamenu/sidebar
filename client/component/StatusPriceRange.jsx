import React from 'react';
import styles from '../sidebar.css';

const price = ['under $10', '$11-30', '$31-60', 'above $61'];

const StatusPriceRange = ({ priceRange }) => {
  const dollarSigns = ['greySign', 'greySign', 'greySign', 'greySign'];
  for (let i = 0; i <= priceRange; i++) {
    dollarSigns[i] = 'greenSign';
  }
  return (
    <tr className={styles.statusIconRow}>
      <td>
        <div className={styles.signShift}>
          {dollarSigns.map((sign, idx) => (
            <b className={styles[sign]} key={idx}>$</b>
          ))}
        </div>
      </td>
      <td className={styles.statusPrice}>
        Price range&nbsp;&nbsp;
        <b id='price-range'>{price[priceRange]}</b>
      </td>
    </tr>
  );
}

export default StatusPriceRange;