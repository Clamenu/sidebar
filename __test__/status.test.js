import React from 'react';
import { shallow } from 'enzyme';
import Status from '../client/component/Status.jsx';
import StatusHours from '../client/component/StatusHours.jsx';
import StatusPriceRange from '../client/component/StatusPriceRange.jsx';
import StatusHealth from '../client/component/StatusHealth.jsx';

describe('Status Window', () => {
  test('should match snapshot for static component', () => {
    const wrapper = shallow(<Status/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Status Hours', () => {
    test('should have 3 parts in 1 line for a continuous time and restaurant is open', () => {
      const wrapper = shallow(<StatusHours hours='7:00 am - 10:00 pm' isOpen={true}/>);
      const tree = wrapper.find('#status-hours').children();
      expect(tree).toHaveLength(3);
    });
    test('should have 4 parts in 2 line for a break mid-day and restaurant is closed', () => {
      const wrapper = shallow(
        <StatusHours hours='7:00 am - 10:00 am, 2:00 pm - 10:00 pm' isOpen={false}/>
      );
      const tree = wrapper.find('#status-hours').children();
      expect(tree).toHaveLength(4);
    });
    test('should say Open now when restaurant is open', () => {
      const wrapper = shallow(
        <StatusHours hours='7:00 am - 10:00 am, 2:00 pm - 10:00 pm' isOpen={true}/>
      );
      const open = wrapper.find('span').text();
      expect(open).toBe('Open now');
    });
    test('should say Closed now when restaurant is closed', () => {
      const wrapper = shallow(<StatusHours hours='7:00 am - 10:00 pm' isOpen={false}/>);
      const close = wrapper.find('span').text();
      expect(close).toBe('Closed now');
    });
  });

  describe('Status Price Range', () => {
    test('should have 3 parts in 1 line for a continuous time and restaurant is open', () => {
      const wrapper = shallow(<StatusPriceRange priceRange={2}/>);
      const price = wrapper.find('#price-range').text();
      expect(price).toBe('$31-60');
    });
  });

  describe('Status Health', () => {
    test('should have expected health score when passed in', () => {
      const wrapper = shallow(<StatusHealth healthScore={92}/>);
      const healthScore = wrapper.find('#health-score').text();
      expect(healthScore).toContain('92 out of 100');
    });
  });
});