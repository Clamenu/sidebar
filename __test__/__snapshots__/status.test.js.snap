// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Window should match snapshot for static component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table>
        <tbody>
          <Connect(StatusHours) />
          <Connect(StatusPriceRange) />
          <Connect(StatusHealth) />
        </tbody>
      </table>,
      "className": "statusContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tbody>
          <Connect(StatusHours) />
          <Connect(StatusPriceRange) />
          <Connect(StatusHealth) />
        </tbody>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(StatusHours) />,
            <Connect(StatusPriceRange) />,
            <Connect(StatusHealth) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table>
          <tbody>
            <Connect(StatusHours) />
            <Connect(StatusPriceRange) />
            <Connect(StatusHealth) />
          </tbody>
        </table>,
        "className": "statusContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tbody>
            <Connect(StatusHours) />
            <Connect(StatusPriceRange) />
            <Connect(StatusHealth) />
          </tbody>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(StatusHours) />,
              <Connect(StatusPriceRange) />,
              <Connect(StatusHealth) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
